src/attribute_type.rs

```diff
@@ -1905,146 +1905,153 @@ fn parse_fmtp(to_parse: &str) -> Result<SdpAttribute, SdpParserInternalError> {
         unknown_tokens: Vec::new(),
     };
 
-    if parameter_token.contains('=') {
-        let parameter_tokens: Vec<&str> = parameter_token.split(';').collect();
-
-        for parameter_token in parameter_tokens.iter() {
-            let name_value_pair: Vec<&str> = parameter_token.splitn(2, '=').collect();
-
-            if name_value_pair.len() != 2 {
-                return Err(SdpParserInternalError::Generic(
-                    "A fmtp parameter must be either a telephone event, a parameter list or
+    for parameter_token in tokens[1..].to_vec() {
+        if parameter_token.contains('=') {
+            // Permit Leading/Trailing/Inner ';' by filtering out empty splits
+            let parameter_tokens: Vec<&str> = parameter_token
+                .split(';')
+                .filter(|token| !token.is_empty())
+                .collect();
+            for parameter_token in parameter_tokens.iter() {
+                let name_value_pair: Vec<&str> = parameter_token.splitn(2, '=').collect();
+                if name_value_pair.len() != 2 {
+                    return Err(SdpParserInternalError::Generic(
+                        "A fmtp parameter must be either a telephone event, a parameter list or
```

Comment:
The content of this error doesn't look right.
If I remember right the formatting of the indented string "a red codec list" also appeared indented in the file_parser output.

Reply:
Yeah, I remember seeing that as well...
Fixed.